### ISSUE TITLE
The model architecture of word2vec

### ISSUE THREAD

**Issue by 1641004802** [piskvorky/gensim#3543]
Excuse me，I want to know that the model architecture of gensim.models.Word2Vec（CBOW）.
It seems that the architecture is not as simple as an embedding layer，a hidden layer（linear）and softmax.

**Comment by gojomo** [comment]
The best reference is the source code itself, available in this repository, which was originally a close port of Google's `word2vec.c` release (though it has since added many other Python/Cython optimizations and additional options). The inner neural network is just a projection/embedding-layer, hidden layer, and output layer which is trained either by negative-sampling (default & most-common) or hierarchical-softmax (*not* plain softmax) - just as in `word2vec.c`. 

Other than that, open-ended discussion is better supported in the project's discussion group, https://groups.google.com/g/gensim , reserving this issue-tracker for bugs/feature-requests.

**Comment by gojomo** [comment]
The best reference is the source code itself, available in this repository, which was originally a close port of Google's `word2vec.c` release (though it has since added many other Python/Cython optimizations and additional options). The inner neural network is just a projection/embedding-layer, hidden layer, and output layer which is trained either by negative-sampling (default & most-common) or hierarchical-softmax (*not* plain softmax) - just as in `word2vec.c`. 

Other than that, open-ended discussion is better supported in the project's discussion group, https://groups.google.com/g/gensim , reserving this issue-tracker for bugs/feature-requests.
